index.bs

```diff
@@ -282,30 +266,164 @@ Loader instances are initially created with the internal slots described in the
+</emu-alg>
+
+The value of the name property of this function is "[Symbol.iterator]".
+
+<h4 id="registry-prototype-get">Registry.prototype.get(key)</h4>
+
+The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Let _entries_ be _registry_.[[Entries]].
+1. Let _pair_ be the entry in _entries_ such that _pair_.[[Key]] is equal to _key_.
+1. If _pair_ exists, then return _pair_.[[Value]]. Otherwise return *undefined*.
+</emu-alg>
+
+<h4 id="registry-prototype-set">Registry.prototype.set(key, entry)</h4>
+
+The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Assert: _entry_ is an initialized RegistryEntry instance.
+1. Let _entries_ be _registry_.[[Entries]].
+1. Let _pair_ be the entry in _entries_ such that _pair_.[[Key]] is equal to _key_.
+1. If _pair_ exists, then throw a new *TypeError*.
+1. Append { [[Key]]: _key_, [[Value]]: _entry_ } to _entries_.
+1. Return _registry_.
+</emu-alg>
 
-<h4 id="registry-GetCurrentStage" aoid="GetCurrentStage">GetCurrentStage(entry)</h4>
+<h4 id="registry-prototype-has">Registry.prototype.has(key)</h4>
+
+The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Let _entries_ be _registry_.[[Entries]].
+1. Let _pair_ be the entry in _entries_ such that _pair_.[[Key]] is equal to _key_.
+1. If _pair_ exist, return *true*. Otherwise return *false*.
+</emu-alg>
+
+<h4 id="registry-prototype-delete">Registry.prototype.delete(key)</h4>
+
+The following steps are taken:
+
+<emu-alg>
+1. Let _registry_ be *this* value.
+1. If Type(_registry_) is not Object, throw a *TypeError* exception.
+1. If _registry_ does not have a [[Entries]] internal slot, throw a *TypeError* exception.
+1. Let _entries_ be _registry_.[[Entries]].
+1. Let _pair_ be the entry in _entries_ such that _pair_.[[Key]] is equal to _key_.
+1. If _pair_ does not exist, then throw a new *TypeError*.
+1. Remove _pair_ from _entries_.
+1. Return *true*.
+</emu-alg>
+
+<h3 id="registry-internal-slots">Properties of Registry Instances</h3>
+
+Registry instances are ordinary objects that inherit properties from the %RegistryPrototype%.
+
+Registry instances are initially created with the internal slots described in the following table:
+
+<table>
+  <thead>
+    <tr>
+      <th>Internal Slot</th>
+      <th>Value Type (<em>non-normative</em>)</th>
+      <th>Description (<em>non-normative</em>)</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>\[[Entries]]</td>
```

Comment:
It seems to be 262 style to have the internal property name be prefixed with the container type (e.g. \[\[MapData]]), so we should have \[\[RegistryEntries]] instead of \[\[Entries]].